### PR TITLE
Canvas Refactor

### PR DESCRIPTION
**Pull request by AnturK**
Makes canvases use TGUI.
Adds painting frames craftable from wood.

Unfortunately having the painted icons in base map is still not really feasible, but this restores at least the basic functionality. I'll probably add persistence for this in another PR later.

Could use better sprites for frames and maybe some more involved color detail overlays.

:cl:
add: Canvases are now available again.
add: Painting frames can be built from wood. Use a pen to name your masterpiece.
/:cl:

dreamseeker_2020-01-16_20-06-41

### CODE REVIEW

**Review by Jalleo — CHANGES_REQUESTED** [review]
Please correct your main post. Its TGUI-next not TGUI

**Comment by optimumtact** [comment]
@Jalleo that is not an appropriate use of the review functionality

File: tgui-next/packages/tgui/interfaces/Canvas.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        height={y_size || 300}
+        {...rest}
+        onClick={e => this.clickwrapper(e)}>
+          Canvas failed to render.
+      </canvas>
+    );
+  }
+}
+
+PaintCanvas.defaultHooks = pureComponentHooks;
```
Comment (stylemistake) [inline tgui-next/packages/tgui/interfaces/Canvas.js]: Can you explain what effect this has on canvas? Perhaps this hides some bug which could be solved in some other way.

File: tgui-next/packages/tgui/interfaces/Canvas.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+export const Canvas = props => {
+  const { act, data } = useBackend(props);
+  return (
+    <Box textAlign="center">
+      <PaintCanvas
+        value={data.grid}
+        onCanvasClick={(x, y) => act("paint", { x: x, y: y })} />
```
Comment (stylemistake) [inline tgui-next/packages/tgui/interfaces/Canvas.js]: You can just write `{ x, y }`

**Review by stylemistake — COMMENTED** [review]
Very nice.

File: tgui-next/packages/tgui/interfaces/Canvas.js (hunk 1 above)
Comment (AnturK) [reply tgui-next/packages/tgui/interfaces/Canvas.js]: I had some issues with canvas smuding when i started and tried this as a fix, forgot to remove this.

**Comment by Farquaar** [comment]
Ah yes, now I can finally build the Space Station 13 Centre for Culture and Arts like I had always dreamed of.

**Review by stylemistake — DISMISSED** [review]

**Comment by Cyberboss** [comment]
Needs rebuild

**Pull request merged**